ecc_mock/chaincode/ecc.go

```diff
@@ -52,103 +51,105 @@ func (t *EnclaveChaincode) Invoke(stub shim.ChaincodeStubInterface) pb.Response
-	// attestationParamsBase64 := stub.GetStringArgs()[0]
 
 	// fetch cc params and host params
+	chaincodeParams, err := ExtractChaincodeParams(stub)
+	if err != nil {
+		shim.Error(err.Error())
+	}
 
-	// create enclave
-
-	// ecall_init(cc params, host params, and attestation params)
-
-	credentials := &protos.Credentials{
-		Attestation: []byte("{\"attestation_type\":\"simulated\",\"attestation\":\"MA==\"}"),
-		SerializedAttestedData: &any.Any{
-			TypeUrl: proto.MessageName(&protos.AttestedData{}),
-			Value: protoutil.MarshalOrPanic(&protos.AttestedData{
-				EnclaveVk: []byte("enclaveVKString"),
-				CcParams: &protos.CCParameters{
-					ChaincodeId: "ercc",
-					Version:     "1.0",
-					ChannelId:   "mychannel",
-					Sequence:    1,
-				},
-				HostParams: &protos.HostParameters{
-					PeerMspId: "org1",
-					// PeerEndpoint:
-				},
-			}),
-		},
+	hostParams, err := extractHostParams(stub)
+	if err != nil {
+		shim.Error(err.Error())
 	}
 
-	credentialBytes, err := proto.Marshal(credentials)
+	attestationParams, err := extractAttestationParams(stub)
 	if err != nil {
 		shim.Error(err.Error())
 	}
 
-	logger.Infof("return some credentials: %s", credentials)
+	credentialsBytes, err := t.enclave.Init(chaincodeParams, hostParams, attestationParams)
+	if err != nil {
+		shim.Error(err.Error())
+	}
 
 	// return credentials
-	return shim.Success(credentialBytes)
+	return shim.Success(credentialsBytes)
 }
 
 func (t *EnclaveChaincode) invoke(stub shim.ChaincodeStubInterface) pb.Response {
-	argsBase64 := stub.GetStringArgs()[1]
-	if argsBase64 == "" {
-		logger.Errorf("empty arguments")
-		return shim.Error("empty arguments")
-	}
-
-	args, err := base64.StdEncoding.DecodeString(argsBase64)
-	if err != nil {
-		logger.Errorf(err.Error())
-		return shim.Error(err.Error())
-	}
 
 	// call enclave
-	var errMsg string
-	responseData, signature, errInvoke := t.enclave.Invoke(args, nil, stub)
+	responseBytes, errInvoke := t.enclave.ChaincodeInvoke(stub)
 	if errInvoke != nil {
-		errMsg = fmt.Sprintf("t.enclave.Invoke failed: %s", errInvoke)
+		errMsg := fmt.Sprintf("t.enclave.ChaincodeInvoke failed: %s", errInvoke)
 		logger.Errorf(errMsg)
-		// likely a chaincode error, so we stil want response go back ...
-	}
-
-	fpcResponse := &utils.Response{
-		ResponseData: responseData,
-		Signature:    signature,
-		PublicKey:    []byte("TODO enclave public key"),
-	}
-	responseBytes, _ := json.Marshal(fpcResponse)
-
-	var response pb.Response
-	if errInvoke == nil {
-		response = pb.Response{
-			Status:  shim.OK,
-			Payload: responseBytes,
-			Message: errMsg,
-		}
-	} else {
-		response = pb.Response{
+		// likely a chaincode error, so we still want response go back ...
+		return pb.Response{
 			Status:  shim.ERROR,
 			Payload: responseBytes,
 			Message: errMsg,
 		}
 	}
-	return response
+
+	return shim.Success(responseBytes)
 }
 
 func (t *EnclaveChaincode) endorse(stub shim.ChaincodeStubInterface) pb.Response {
```

Comment:
this is definitely a good start for the new validation flow. We need a plan to bring this to ECC.